### PR TITLE
Add leaflet-providers plugin

### PR DESCRIPTION
**Pull request by jcheng5**
Example:

```r
leaflet() %>% addProviderTiles("Stamen.Toner")
```

### CODE REVIEW

**Comment by jcheng5** [comment]
This also sets a precedent for integrating Leaflet plugins.

**Comment by wch** [comment]
FYI, to get this through CRAN these days, the DESCRIPTION file will need a `c(person(...), person(...))` format, and you'll need a LICENSE file that has the licenses of all included components. See [shiny](https://github.com/rstudio/shiny) and [shinydashboard](https://github.com/rstudio/shinydashboard) for examples. They're also very particular about all contributors being credited in the DESCRIPTION if possible.

**Comment by jcheng5** [comment]
Thanks @wch, made those changes.

**Comment by jcheng5** [comment]
@yihui I believe this is ready to merge now.

**Comment by yihui** [comment]
Thanks! This looks great!

File: DESCRIPTION [hunk 1]
```diff
@@ -9,6 +9,7 @@ Authors@R: c(
       person("Hadley", "Wickham", role = c("ctb")),
       person("jQuery contributors", role = c("ctb", "cph"), comment = "jQuery library"),
       person("Leaflet contributors", role = c("ctb", "cph"), comment = "Leaflet library"),
+      person("Leaflet Providers contributors", role = c("ctb", "cph"), comment = "Leaflet Providers plugin"),
```
Comment (wch) [inline DESCRIPTION]: I think that they should be listed as `ctb` only. See the LICENSE:
  https://github.com/Leaflet/Leaflet/blob/master/LICENSE

Comment (jcheng5) [inline DESCRIPTION]: I don't understand... What about the license makes them not cph?

Comment (wch) [inline DESCRIPTION]: Oops, I was looking at Leaflet, not Leaflet providers. But the point still stands, for the previous line.

Also, I think that Vladimir Agafonkin should be `cph` and `ctb`, and CloudMade should be as `cph`.

Comment (wch) [inline DESCRIPTION]: For Leaflet, the contributors aren't copyright holders, according to the LICENSE file. (Though I suppose that if they haven't signed a CLA, it's possible the LICENSE file doesn't correctly reflect the copyright holders.)

Comment (jcheng5) [inline DESCRIPTION]: Oh, I see what you're saying. I'll make the change on master. Thanks

**Pull request merged**